packages/astro/src/vite-plugin-markdown/content-entry-type.ts

```diff
@@ -15,4 +16,22 @@ export const markdownContentEntryType: ContentEntryType = {
 	},
 	// We need to handle propagation for Markdown because they support layouts which will bring in styles.
 	handlePropagation: true,
+
+	async getRenderFunction(settings) {
+		const processor = await createMarkdownProcessor(settings.config.markdown);
```

Comment:
Wow, this is simpler than I expected. This works great.

Also want to put this on your radar: we have a specialized field called `remarkPluginFrontmatter` that includes any data that's computed by a remark plugin. [A common use case is "reading time,"](https://docs.astro.build/en/recipes/reading-time/) which extracts text nodes from remark to compute minutes read. Is this covered by `result.metadata` potentially? Or is this a consideration we would need to tackle separately, if we support it?